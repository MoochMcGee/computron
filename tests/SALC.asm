[bits 16]

xor al, al
stc
salc
clc
salc

db 0xf1
