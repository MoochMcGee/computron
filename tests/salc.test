asm
	stc
	salc
_asm

result
	CF=1
	AX=00FF
_result
