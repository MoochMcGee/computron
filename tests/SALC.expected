1000:00000000 30 EAX=00000000 EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1000 ES=0000 SS=9000 FS=0000 GS=0000 C=0 P=0 A=0 Z=0 S=0 I=1 D=0 O=0 A16 O16 X16
1000:00000002 F9 EAX=00000000 EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1000 ES=0000 SS=9000 FS=0000 GS=0000 C=0 P=1 A=0 Z=1 S=0 I=1 D=0 O=0 A16 O16 X16
1000:00000003 D6 EAX=00000000 EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1000 ES=0000 SS=9000 FS=0000 GS=0000 C=1 P=1 A=0 Z=1 S=0 I=1 D=0 O=0 A16 O16 X16
1000:00000004 F8 EAX=000000FF EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1000 ES=0000 SS=9000 FS=0000 GS=0000 C=1 P=1 A=0 Z=1 S=0 I=1 D=0 O=0 A16 O16 X16
1000:00000005 D6 EAX=000000FF EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1000 ES=0000 SS=9000 FS=0000 GS=0000 C=0 P=1 A=0 Z=1 S=0 I=1 D=0 O=0 A16 O16 X16
1000:00000006 F1 EAX=00000000 EBX=00000000 ECX=00000000 EDX=00000000 ESP=00001000 EBP=00000000 ESI=00000000 EDI=00000000 CR0=00000000 A20=0 DS=1000 ES=0000 SS=9000 FS=0000 GS=0000 C=0 P=1 A=0 Z=1 S=0 I=1 D=0 O=0 A16 O16 X16
